torch/nn/modules/transformer.py

```diff
@@ -17,13 +17,17 @@
 
 def _generate_square_subsequent_mask(
         sz: int,
-        device: torch.device = torch.device(torch._C._get_default_device()),  # torch.device('cpu'),
```

Comment:
These functions should not be called as the default (and it doesn't make sense either to have them here because after importing changing the default device/dtype does not change it in the function

```
>>> import torch
>>> from torch.nn.modules.transformer import _generate_square_subsequent_mask
>>> _generate_square_subsequent_mask(2)
tensor([[0., -inf],
        [0., 0.]])
>>> torch.set_default_device('cuda')
# still on CPU
>>> _generate_square_subsequent_mask(2)
tensor([[0., -inf],
        [0., 0.]])
```

Changing the default to `torch.device('cpu') also seems to initialize numpy so I think setting it to `None` fixes the error in the most BC-compatible way